test/exposure.jl

```diff
@@ -0,0 +1,129 @@
+using Dispersal
+using DynamicGrids, Unitful, Dates
+using DimensionalData
+using Test
+
+@testset "Test exposure population" begin
+
+    init = (
+        pesticide =  [0.0 0.0 0.0; 
+                    0.0 0.0 0.0;
+                    0.0 0.0 0.0],
+        population = [100000.0 100.0 0.0;
+                    0.0 1000.0 0.0;
+                    0.0 0.0 100000.0], )
+
+    crop = [0.0 1.0 1.0; 
+    0.0 10.0 0.0;
+    0.0 0.0 10.0]
+        
+    output = ArrayOutput(init; tspan=1:3,  aux=(crop=crop,))
+    ruleTreatment = Threshold_Exposure{Tuple{:pesticide,:population}, :pesticide}(crop=Aux(:crop), pulseLevel=20, popThreshold = 1000)
+    rule = Ruleset(ruleTreatment)
+    sim!(output, rule)
+    @test output[Ti(1)] == (
+        pesticide = [0.0 0.0 0.0; 0.0 0.0 0.0; 0.0 0.0 0.0],
+        population = [100000.0 100.0 0.0; 0.0 1000.0 0.0; 0.0 0.0 100000.0])
+    @test output[Ti(2)] == (
+        pesticide = [0.0 0.0 0.0; 0.0 20.0 0.0; 0.0 0.0 20.0],
+        population = [100000.0 100.0 0.0; 0.0 1000.0 0.0; 0.0 0.0 100000.0])
+    @test output[Ti(3)] == (
+        pesticide = [0.0 0.0 0.0; 0.0 32.13061319425267 0.0; 0.0 0.0 32.13061319425267],
+        population = [100000.0 100.0 0.0; 0.0 1000.0 0.0; 0.0 0.0 100000.0])
+
+    # If cropping change allong time
+    crop = reshape( # TODO: something that change with time
+        repeat([0.0 1.0 1.0; 
+                0.0 10.0 0.0;
+                0.0 0.0 10.0], outer=(1,3) ), (3,3,3))
+    cropDD = DimArray(crop, (X, Y, Ti));
+
+    output = ArrayOutput(init; tspan=1:3,  aux=(cropDD=cropDD,))
+    
+    ruleTreatment = Threshold_Exposure{Tuple{:pesticide,:population}, :pesticide}(crop=Aux(:cropDD), pulseLevel=20, popThreshold = 1000)
+
+    rule = Ruleset(ruleTreatment)
+    sim!(output, rule)
+    @test output[Ti(1)] == (
+        pesticide = [0.0 0.0 0.0; 0.0 0.0 0.0; 0.0 0.0 0.0],
+        population = [100000.0 100.0 0.0; 0.0 1000.0 0.0; 0.0 0.0 100000.0])
+    @test output[Ti(2)] == (
+        pesticide = [0.0 0.0 0.0; 0.0 20.0 0.0; 0.0 0.0 20.0],
+        population = [100000.0 100.0 0.0; 0.0 1000.0 0.0; 0.0 0.0 100000.0])
+    @test output[Ti(3)] == (
+        pesticide = [0.0 0.0 0.0; 0.0 32.13061319425267 0.0; 0.0 0.0 32.13061319425267],
+        population = [100000.0 100.0 0.0; 0.0 1000.0 0.0; 0.0 0.0 100000.0])
+
+end   
+
+##################
+# GENERIC ROTATION
+##################
+
+@testset "rotation structure" begin
+    RS(x,y) = Dispersal.RotationStruct(x,y)
```

Comment:
This uppercase method is fine, as its actually a constructor and it suggests that.